test/langtools/jdk/javadoc/doclet/testMarkdown/TestMarkdownCodeBlocks.java

```diff
@@ -486,4 +486,61 @@ public int hashCode() {
                     <dd><code><a href="NullPointerException.html" title="class in p">NullPointerException</a></code> - if other is <code>null</code></dd>
                     </dl>""");
     }
+
+    @Test
+    public void testLeadingCodeBlock(Path base) throws Exception {
+        Path src = base.resolve("src");
+        tb.writeJavaFiles(src,
+                """
+                    package p;
+                    ///     Leading code block
+                    /// Lorum ipsum.
```

Comment:
Yes, you typically need a blank line _before_ an indented code block, to avoid being treated as a continuation of a preceding paragraph, but "unindenting" is enough to end an indented code block.